tools/tsp-client/src/index.ts

```diff
@@ -234,17 +224,17 @@ async function main() {
             throw new Error("Commit SHA is required when specifying `--repo`, please specify a commit using `--commit`");
         }
         if (options.commit) {
-          let [ directory, commit, repo, additionalDirectories ] = await readTspLocation(rootUrl);
-          commit = options.commit ?? commit;
-          repo = options.repo ?? repo;
-          await writeFile(path.join(rootUrl, "tsp-location.yaml"), `directory: ${directory}\ncommit: ${commit}\nrepo: ${repo}\nadditionalDirectories: ${additionalDirectories}`);
+          let tspLocation: TspLocation = await readTspLocation(rootUrl);
```

Comment:
if you are not reassigning tspLocation it is usually recommended to make it a cosnt

Reply:
even if you modify some of the properties